### PR TITLE
PLG-783 simplify useFetchProductScores

### PR DESCRIPTION
**Pull request by FSAkeneo**
Simplification/fix of useFetchProductQualityScore.

- [x] Tests

### CODE REVIEW

**Comment by FSAkeneo** [comment]
/!\ until PLG-822 is merged into master, this PR includes its commits

File: src/Akeneo/Pim/Automation/DataQualityInsights/tests/front/unit/infrastructure/hooks/ProductEditForm/useFetchQualityScore.unit.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          fetchQualityScoreMock.mockImplementationOnce(async () => {
+            await sleep(10);
+            throw Error('some error message');
+          });
+
+          await act(() => result.current.fetcher());
+
+          await waitFor(() => {
+            if (result.current.outcome.status !== 'failed') {
+              console.log(result.current.outcome);
```
Comment (BryanS29-akeneo) [inline src/Akeneo/Pim/Automation/DataQualityInsights/tests/front/unit/infrastructure/hooks/ProductEditForm/useFetchQualityScore.unit.ts:134]: Did you forget this console.log ?

**Review by BryanS29-akeneo — APPROVED** [review]

**Review by LaurentPetard — APPROVED** [review]

**Review by LaurentPetard — CHANGES_REQUESTED** [review]
When a product has been updated from a bulk action, import, or API, we don't know how long the user will wait to see the new score. So in theses case we display the "In progress" badge.
The flashing "ghost" score should only be displayed when a product has been updated from the PEF (i.e. when the HTTP request to re-evaluate the product is in progress)

How can I change your own comment ??? (fsakeneo)

**Comment by FSAkeneo** [comment]
> When a product has been updated from a bulk action, import, or API, we don't know how long the user will wait to see the new score. So in theses case we display the "In progress" badge. The flashing "ghost" score should only be displayed when a product has been updated from the PEF (i.e. when the HTTP request to re-evaluate the product is in progress)

I've made some changes, it should be fine now when we don't have the score and we did not save juste before (the "long pending after mass update" scenario).
I also noticed than when we switch from PEF to product grid while score retrievel retrying is in progress it wasn't (and could not have been) stopped. Fixed that I think.

**Review by LaurentPetard — APPROVED** [review]

**Pull request merged**